src/connection/host_lookup.c

```diff
@@ -0,0 +1,635 @@
+	{
+	  return NULL;
+	}
+
+    }
+
+  /*Look up in the user_host_Map */
+  if ((lookup_type == HOSTNAME_TO_IPADDR) && (user_host_Map.find (hostname) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (hostname)->second];
+    }
+  else if ((lookup_type == IPADDR_TO_HOSTNAME) && (user_host_Map.find (addr_trans_ch_buf) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (addr_trans_ch_buf)->second];
+    }
+  else
+    {
+      if (lookup_type == HOSTNAME_TO_IPADDR)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_CANT_LOOKUP_INFO, 1, hostname);
+	  fprintf (stdout, "%s\n", er_msg ());
+	}
+      return NULL;
+    }
+
+  return hp;
+}
+
+static int
+load_hosts_file ()
+{
+  FILE *fp;
+  char file_line[LINE_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token, temp_token[LINE_BUF_SIZE + 1];
+  char *save_ptr_strtok;
+  /*delimiter */
+  const char *delim = " \t\n";
+  char ipaddr[IPADDR_LEN];
+  char hostname[HOSTNAME_LEN];
+  int cache_idx = 0, temp_idx;
+
+  char addr_trans_ch_buf[IPADDR_LEN];
+  struct in_addr addr_trans;
+
+  /*True, when the string token has hostname, otherwise, string token has IP address */
+  bool hostent_flag;
+
+  memset (file_line, 0, LINE_BUF_SIZE);
+
+  hosts_conf_dir = envvar_confdir_file (host_conf_file_full_path, PATH_MAX, user_hosts_File);
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_IO_MOUNT_FAIL, 1, host_conf_file_full_path);
+      fprintf (stdout, "%s\n", er_msg ());
+      return LOAD_FAIL;
+    }
+
+  while (fgets (file_line, LINE_BUF_SIZE, fp) != NULL)
+    {
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      hostent_flag = INSERT_IPADDR;
+
+      do
+	{
+	  if (token == NULL || *token == '#')
+	    {
+	      break;
+	    }
+	  if (hostent_flag == INSERT_IPADDR)
+	    {
+	      strcpy (temp_token, token);
+	      if (ip_format_check (temp_token) == false)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_IP_ADDR_INVALID_FORMAT, 1, token);
+		  fprintf (stdout, "%s\n", er_msg ());
+
+		  user_host_Map.clear ();
+
+		  fclose (fp);
+
+		  return LOAD_FAIL;
+		}
+	      else
+		{
+		  strncpy (ipaddr, token, IPADDR_LEN);
+		  ipaddr[IPADDR_LEN - 1] = '\0';
+
+		  hostent_flag = INSERT_HOSTNAME;
+		}
+	    }
+	  else
+	    {
+	      if (strlen (token) > HOSTNAME_LEN - 1)
+		{
+		  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_HOST_NAME_TOO_LONG, 1, token);
+		  fprintf (stdout, "%s\n", er_msg ());
+
+		  user_host_Map.clear ();
+
+		  fclose (fp);
+
+		  return LOAD_FAIL;
+		}
+	      else
+		{
+		  strcpy (hostname, token);
+		}
+	    }
+	}
+      while ((token = strtok_r (NULL, delim, &save_ptr_strtok)) != NULL);
+
+      if (strcmp ("\0", hostname) && strcmp ("\0", ipaddr))
```

Comment:
If you can check it simply, it would be better to avoid using functions.
```suggestion
      if (hostname[0] && ipaddr[0])
```

Reply:
Thank you! It looks so good and efficient!